tests/Config/ConfigResolverTest.php

```diff
@@ -0,0 +1,299 @@
+            $this->createStub(Twigcs\Container::class),
+            self::temporaryDirectory(),
+            []
+        );
+
+        $this->expectException(Twigcs\Config\InvalidConfigurationException::class);
+
+        $configResolver->getConfigFile();
+    }
+
+    /**
+     * @dataProvider provideDefaultConfigFileName
+     */
+    public function testGetConfigFileReturnsConfigFileWhenConfigOptionHasNotBeenSpecifiedAndDefaultConfigFileExistAndReturnsConfig(string $defaultConfigFileName): void
+    {
+        $defaultConfigFilePath = sprintf(
+            '%s/%s',
+            self::temporaryDirectory(),
+            $defaultConfigFileName
+        );
+
+        $defaultConfigFileContent = <<<PHP
+<?php
+
+return FriendsOfTwig\Twigcs\Config\Config::create();
+PHP;
+
+        self::fileSystem()->dumpFile(
+            $defaultConfigFilePath,
+            $defaultConfigFileContent
+        );
+
+        $configResolver = new Twigcs\Config\ConfigResolver(
+            $this->createStub(Twigcs\Container::class),
+            self::temporaryDirectory(),
+            []
+        );
+
+        self::assertSame($defaultConfigFilePath, $configResolver->getConfigFile());
+    }
+
+    /**
+     * @dataProvider provideDefaultConfigFileNameAndAvailableDefaultConfigFileNames
+     *
+     * @param array<int, string> $availableDefaultConfigFileNames
+     */
+    public function testGetConfigFileReturnsConfigFileWhenConfigOptionHasNotBeenSpecifiedAndMultipleDefaultConfigFilesExistsAndReturnConfigs(
+        string $defaultConfigFileName,
+        array $availableDefaultConfigFileNames
+    ): void {
+        $defaultConfigFilePath = sprintf(
+            '%s/%s',
+            self::temporaryDirectory(),
+            $defaultConfigFileName
+        );
+
+        $defaultConfigFileContent = <<<PHP
+<?php
+
+return FriendsOfTwig\Twigcs\Config\Config::create();
+PHP;
+
+        foreach ($availableDefaultConfigFileNames as $availableDefaultConfigFileName) {
+            $availableDefaultConfigFilePath = sprintf(
+                '%s/%s',
+                self::temporaryDirectory(),
+                $defaultConfigFileName
+            );
+
+            self::fileSystem()->dumpFile(
+                $availableDefaultConfigFilePath,
+                $defaultConfigFileContent
+            );
+        }
+
+        $configResolver = new Twigcs\Config\ConfigResolver(
+            $this->createStub(Twigcs\Container::class),
+            self::temporaryDirectory(),
+            []
+        );
+
+        self::assertSame($defaultConfigFilePath, $configResolver->getConfigFile());
+    }
+
+    /**
+     * @return array<int, array{0: string, 1: array<int, string>}>
+     */
+    public function provideDefaultConfigFileNameAndAvailableDefaultConfigFileNames(): array
+    {
+        return [
+            [
+                '.twig_cs',
+                [
+                    '.twigcs',
+                    '.twigcs.dist',
+                ],
+            ],
```

Comment:
@Ciloe @OwlyCode @yguedidi

In the context of #247, we could add more data sets here to ensure that configuration files are resolved in the specified order - what do you think?

Reply:
sure, I'll rework my PR when this is merged